sharkfuser/include/fusilli/attributes/tensor_attributes.h

```diff
@@ -467,21 +548,65 @@ class TensorAttr {
     return expectedStride == stride_;
   }
 
-  // Convert logical dims + stride into physical dims
-  //  dims [N, C, H, W] + strideOrder [3, 2, 1, 0] -> [N, C, H, W]
-  //  dims [N, C, H, W] + strideOrder [3, 0, 2, 1] -> [N, H, W, C]
-  std::vector<int64_t> getPhysicalDim() const {
+  // Check if the stride pattern is valid and can represent a physical tensor.
+  // A valid stride pattern must satisfy:
+  // 1. Each stride is a product of dimensions in faster-changing positions
+  // 2. The strides are consistent with some permutation of dimensions
+  bool hasValidPhysicalRepresentation() const {
     size_t numDims = dim_.size();
+    if (numDims == 0)
+      return true;
+    if (numDims != stride_.size())
+      return false;
```

Comment:
Could you please move this check up? It seems like it should be before empty tensor validation